Assets/Scripts/Boat/BoatMovement.cs

```diff
@@ -0,0 +1,129 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Unity.Mathematics;
+using UnityEngine;
+
+public class BoatMovement : MonoBehaviour
+{
+    [Tooltip("Boat Movement Speed in m/s")]
+    public float maxSpeed;
+    
+    [ReadOnly(true)][Tooltip("The current speed the boat is moving in m/s")]
+    public float currentSpeed;
+
+    [Tooltip("Boat Acceleration in m/s")]
+    public float acceleration;
+    
+    [Tooltip("Boat rotational speed in degrees per second.")]
+    public float rotationSpeed;
+    
+    [Tooltip("Boat rotational limits in degrees. X: Clockwise Limit, Y: AntiClockwise Limit")]
+    public Vector2 rotationLimits;
+
+    public Vector3 currentDirection;
+    [SerializeField] private bool isMoving = false;
+    
+    public KeyCode rotateLeftKeyCode;
```

Comment:
Are these KeyCode variables supposed to be serialized?

Reply:
Don't need to be because they are public. Still visible in the editor. Keycodes are a temporary solution until we get the InputSystem in.